File: PDOO/Napakalaki-Java/src/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.*;

/**
 *
 * @author Cervi
 */
public class CultistPlayer extends Player{
    
    static private int totalCultistPlayers = 0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        this.myCultistCard=c;
        totalCultistPlayers++;
    }
    
    
    public int getCombatLevel(){
        int returnLevel=(int) (super.getCombatLevel()+(super.getCombatLevel()*0.2));
        returnLevel+=myCultistCard.getGainedLevels()*totalCultistPlayers;
        return returnLevel;
    }
    
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    
    private Treasure giveMeATreasure(){
        Random rand = new Random(); 
        return visibleTreasures.get(rand.nextInt(visibleTreasures.size()-1));
    }
    
    private boolean canYouGiveMeATreasure(){
        return (!visibleTreasures.isEmpty() || visibleTreasures != null);
    }
    
    public int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
}

File: PDOO/Napakalaki-Ruby/pkg/Napakalaki-Ruby-0.0.1/lib/CultistPlayer.rb
#encoding: utf-8

  require 'Player.rb'
  require 'Cultist.rb'

module NapakalakiGame


class CultistPlayer < Player
  
  @@totalCultistPlayers = 0
  
  def initialize(p, c)
    super(p)
    @myCultistCard = c
    @@totalCultistPlayers += 1
  end
  
  def getCombatLevel()
    nivel = @level
    nivel = (nivel + (nivel*0.2))
    nivel += @myCultistCard.gainedLevels*@@totalCultistPlayers
    return nivel.to_i
  end
  
  def getOponentLevel(m)
    return m.getCombatLevelAgainstCultistPlayer
  end
  
  def shouldConvert()
    return false
  end
  
  def giveMeATreasure()
    return @visibleTreasures[rand(@visibleTreasures.size-1)]
  end
  
  def canYouGiveMeATreasure()
    if( @visibleTreasures != nil || ! @visibleTreasures.empty?)
      return true
    end
    return false
  end
  
  def getTotalCultistPlayers
    return @@totalCultistPlayers
  end
 end
end